addons/account/tests/test_portal_attachment.py

```diff
@@ -210,13 +214,17 @@ def test_01_portal_attachment(self):
             url=f'{self.invoice_base_url}/mail/chatter_post',
             json={
                 'params': {
-                    'res_model': self.out_invoice._name,
-                    'res_id': self.out_invoice.id,
-                    'message': "test message 1",
-                    'attachment_ids': [attachment.id],
-                    'attachment_tokens': [attachment.access_token],
+                    'thread_model': self.out_invoice._name,
+                    'thread_id': self.out_invoice.id,
+                    'post_data': {
+                        'body': "test message 1",
+                        'attachment_ids': [attachment.id],
+                        'attachment_tokens': [attachment.access_token],
+                        'portal_security': {
```

Comment:
I don't think this key should be inside `post_data`, why not just keep `token` as a stand-alone key?

Reply:
We have `token`, `hash`, `pid` and `project_sharing_id` which are used as the security parameters in posting message on portal. So I added the `portal_security` in post_data. `post_data` is the only parameter that we have in the `message/post` (backend) that I can pass the security data through it unless we want to add a `kwargs` to its params.

Reply:
`post_data` are supposed to be only for params forwarded to `message_post`

Keys that are used differently (id and model to reference the thread, security things, ...) should be kept separate.

Reply:
Note that on the same principle, `attachment_tokens` should be moved out as well (even in the current method)

Reply:
> Note that on the same principle, `attachment_tokens` should be moved out as well (even in the current method) and also `canned_response_ids`, ... all of those for which we do a ` post_data.pop`.

Do we want to change this in a new PR before proceeding this PR or I can make the changes related to portal for now? 
I can put the security data in a parameter as `security` instead of `post_data` but if we want to reduce the diff in the chatter PR, I need to know about the controller arguments (`mail_message_post`) that we want at the end.